extensions/spring-web/deployment/src/main/java/io/quarkus/spring/web/deployment/SpringWebProcessor.java

```diff
@@ -190,6 +195,35 @@ public void accept(ResteasyDeployment resteasyDeployment) {
         reflectiveClass.produce(new ReflectiveClassBuildItem(true, false, false, SpringResourceBuilder.class.getName()));
     }
 
+    /**
+     * Make sure the controllers have the proper annotation and warn if not
+     */
+    private void validateControllers(BeanArchiveIndexBuildItem beanArchiveIndexBuildItem) {
+        Collection<AnnotationInstance> annotations = beanArchiveIndexBuildItem.getIndex().getAnnotations(REQUEST_MAPPING);
+        Set<DotName> classesWithoutRestController = new HashSet<>();
+        for (AnnotationInstance annotation : annotations) {
+            ClassInfo targetClass;
+            if (annotation.target().kind() == AnnotationTarget.Kind.CLASS) {
+                targetClass = annotation.target().asClass();
+            } else if (annotation.target().kind() == AnnotationTarget.Kind.METHOD) {
+                targetClass = annotation.target().asMethod().declaringClass();
+            } else {
+                continue;
+            }
+
+            if (targetClass.classAnnotation(REST_CONTROLLER_ANNOTATION) == null) {
+                classesWithoutRestController.add(targetClass.name());
+            }
+        }
+
+        if (!classesWithoutRestController.isEmpty()) {
+            for (DotName dotName : classesWithoutRestController) {
+                LOGGER.warn("Class '" + dotName
+                        + "' uses '@RequestMapping' but was not annotated with '@RestContoller' and will therefore be ignored.");
```

Comment:
Sounds like we should fail the build instead? Thinking how confused would I be if the app started and the endpoints would not be there.

Reply:
Yeah, we talked about that with @kamesh and decided that a warning is best for now